sql/sqlbase/structured.go

```diff
@@ -725,6 +725,87 @@ func (desc *TableDescriptor) Validate() error {
 	return desc.Privileges.Validate(desc.GetID())
 }
 
+// FamilyHeuristicMaxBytes is the maximum total byte size of columns that the
+// current heuristic will assign to a family.
+const FamilyHeuristicMaxBytes = 256
+
+// columnValueEncodedSize returns the maximum encoded size of the given column
+// type using the "value" encoding. It accounts for the tag, but assumes the tag
+// is one byte, which is true when the column id part is small (i.e., less than
+// 8). If the size is unbounded, false is returned.
+func columnValueEncodedSize(colType ColumnType) (int, bool) {
+	switch colType.Kind {
+	case ColumnType_BOOL:
+		// BOOLs are stored in the tag byte.
+		return 1, true
```

Comment:
Can all of these be expressed through actual computations (for example built at init time)?